code/modules/antagonists/cult/cult_items.dm

```diff
@@ -106,6 +108,129 @@ Striking a noncultist, however, will tear their flesh."}
+	block_chance = 55
+	wound_bonus = -25
+	bare_wound_bonus = 30
+	free_use = TRUE
+	light_color = COLOR_BLACK
+	light_system = OVERLAY_LIGHT
+	light_range = 4
+	// holder for the actual action when created.
+	var/datum/action/cooldown/spell/path_wielder_action
+	var/mob/living/trapped_entity
+	var/heretic_path
+
+/obj/item/melee/cultblade/haunted/Initialize(mapload, mob/soul_to_bind, mob/awakener, no_binding)
+	. = ..()
+
+	if(!no_binding || mapload)
+		bind_soul(soul_to_bind, awakener)
+	ADD_TRAIT(src, TRAIT_CASTABLE_LOC, INNATE_TRAIT)
+	AddElement(/datum/element/heretic_focus)
+
+/obj/item/melee/cultblade/haunted/proc/bind_soul(mob/soul_to_bind, mob/awakener)
+
+	var/datum/mind/trapped_mind = soul_to_bind.mind
+
+	if(soul_to_bind)
+		AddComponent(/datum/component/spirit_holding,\
+			soul_to_bind = soul_to_bind.mind,\
+			awakener = awakener,\
+			allow_renaming = FALSE,\
+		)
+
+	// Get the heretic's new body and antag datum.
+	trapped_entity = trapped_mind.current
+	var/datum/antagonist/heretic/heretic_holder = IS_HERETIC(trapped_entity)
+	if(!heretic_holder)
+		CRASH("[soul_to_bind] not a heretic on the heretic soul blade.")
+
+	// Set the sword's path for spell selection.
+	heretic_path = heretic_holder.heretic_path
+
+	// Copy the objectives to keep for roundend, remove the datum as neither us nor the heretic need it anymore
+	var/list/copied_objectives = heretic_holder.objectives.Copy()
+	trapped_entity.mind.remove_antag_datum(/datum/antagonist/heretic)
+
+	// Add the fallen antag datum, give them a heads-up of what's happening.
+	var/datum/antagonist/soultrapped_heretic/fallen = trapped_entity.mind.add_antag_datum(/datum/antagonist/soultrapped_heretic)
+	fallen.objectives = copied_objectives
+	to_chat(trapped_entity, span_alert("You've been sacrificed to the Enemy, and trapped inside a haunted blade! While you cannot escape, you may decide by yourself to be a nuisance with the few abilities you still have, or aid whoever wields it."))
+	// Unrelated bug: Objectives dont show up on the you are X thing as they are varsetted post-facto
+
+	// Assigning the spells to give to the wielder and spirit.
+	// Let them cast the given spell.
+	ADD_TRAIT(trapped_entity, TRAIT_ALLOW_HERETIC_CASTING, INNATE_TRAIT)
+
+	var/list/path_spells = GLOB.heretic_paths_to_haunted_sword_abilities[heretic_path]
+
+	var/list/wielder_spells = list(path_spells[WIELDER_SPELL])
+	var/list/sword_spells = list(path_spells[SWORD_SPELL])
+
+	name = path_spells[SWORD_PREFIX] + " " + name
+
+	// Granting the spells. The sword spirit gains it outright, while it's just defined for wielders to be added on pickup.
+
+	if(length(sword_spells))
+		for(var/datum/action/cooldown/spell/sword_spell as anything in sword_spells)
+			sword_spell = new sword_spell(trapped_entity)
+			sword_spell.Grant(trapped_entity)
+
+	if(length(wielder_spells))
+		for(var/datum/action/cooldown/spell/wielder_spell as anything in wielder_spells)
+			path_wielder_action = new wielder_spell(src)
```

Comment:
`for(var/thing in null)` and `for(var/thing in list())` are both valid; they do nothing. You don't need these length checks.